File: fib.jl
function fib(n::Int)
    if n == 1 || n == 2
        return 1
    end
    a, b = 1, 1
    for _ in 3:n
        a, b = b, a + b
    end
    return b
end

val = fib(60)
println(val)

File: fib.f90
program fibonacci
    implicit none
    integer, parameter :: int128 = selected_int_kind(38)
    integer :: i
    integer(kind=int128) :: a, b, next, val
    
    val = fib(60)
    write(*, '(I0)') val
    
contains
    function fib(n) result(res)
        integer, intent(in) :: n
        integer(kind=int128) :: res
        integer(kind=int128) :: a, b, next
        integer :: i
        
        if (n == 1 .or. n == 2) then
            res = 1
            return
        end if
        
        a = 1
        b = 1
        
        do i = 3, n
            next = a + b
            a = b
            b = next
        end do
        
        res = b
    end function fib
end program fibonacci 